### ISSUE TITLE
Is it possible to have Federated Learning on Cloud-Edge?

### ISSUE THREAD

**Issue by Martiniann** [adap/flower#430]
Hi everyone,

Currently I am working on a school project about federated learning and came across your framework during exploratory analysis. My project should utilize federated learning in this manner - I have an aggregation server (let's say in a cloud). I want this server to provide model to my 2 Raspberry PIs. These two RPIs would then train the model on a local data for x epochs and provide the trained models/gradients back to the global server. On this server, the results would be federated averaged and new model would be sent to the PIs. Is such a workflow possible with your framework? If so, could you provide me a hint?

Thank you,
Best regards

**Comment by danieljanes** [comment]
Hi @Martiniann ,

yes, this is possible!

Let's assume you want to do something similar to our Keras Quickstart: `https://flower.dev/quickstart_keras.html`

In your case you would:
1. Start the server on you cloud machine `$ python server.py`
2. Start one client on your Raspberry Pi: `$ python client.py`
3. Start the second client on your other Raspberry Pi: `$ python server.py`

Just make sure that (a) the server is reachable from your Raspberry Pi, and (b) pass the right IP/port to the `start_keras_client` function:

```python
MY_SERVER_ADDRESS = "[::]:8080"  # IPv6 address and port
fl.client.start_keras_client(server_address=MY_SERVER_ADDRESS, client=client)
```

**Comment by danieljanes** [comment]
Hi @Martiniann,

our latest blog post show how to do this in great detail:
https://flower.dev/blog/2020-12-16-running_federated_learning_applications_on_embedded_devices_with_flower

There's also a fully working code example available: https://github.com/adap/flower/tree/main/examples/embedded_devices

**Comment by Martiniann** [comment]
Thank you very much for your response. I will definitely look into it.
